allennlp/data/dataset_readers/semantic_role_labelling.py

```diff
@@ -0,0 +1,228 @@
+                            # or outside a particular span.
+                            predicate_argument_labels = [[] for _ in conll_components[11:-1]]
+                            current_span_label = [None for _ in conll_components[11:-1]]
+
+                        num_annotations = len(predicate_argument_labels)
+                        is_verbal_predicate = False
+                        # Iterate over all verb annotations for the current sentence.
+                        for annotation_index in range(num_annotations):
+                            annotation = conll_components[11 + annotation_index]
+                            label = annotation.strip("()*")
+
+                            if "(" in annotation:
+                                # Entering into a span for a particular semantic role label.
+                                # We append the label and set the current span for this annotation.
+                                bio_label = "B-" + label
+                                predicate_argument_labels[annotation_index].append(bio_label)
+                                current_span_label[annotation_index] = label
+
+                            elif current_span_label[annotation_index] is not None:
+                                # If there's no '(' token, but the current_span_label is not None,
+                                # then we are inside a span.
+                                bio_label = "I-" + current_span_label[annotation_index]
+                                predicate_argument_labels[annotation_index].append(bio_label)
+                            else:
+                                # We're outside a span.
+                                predicate_argument_labels[annotation_index].append("O")
+
+                            # Exiting a span, so we reset the current span label for this annotation.
+                            if ")" in annotation:
+                                current_span_label[annotation_index] = None
+                            # If any annotation contains this word as a verb predicate,
+                            # we need to record its index. This also has the side effect
+                            # of ordering the verbal predicates by their location in the
+                            # sentence, automatically aligning them with the annotations.
+                            if "(V" in annotation:
+                                is_verbal_predicate = True
+
+                        if is_verbal_predicate:
+                            verbal_predicates.append(word_index)
+        return Dataset(instances)
+
+    @classmethod
+    def from_params(cls, params: Params):
+        """
+        Parameters
+        ----------
+        filename : ``str``
+        token_indexers: ``List[Params]``, optional
+        """
+        filename = params.pop('filename')
+        token_indexers = {}
```

Comment:
We should change `TokenIndexer.from_params()` to assume we have a JSON dict, and return the dictionary, so we don't have to repeat this anywhere.  I'll take care of that in a separate PR.